# To test the hypothesis that the First_Baseman are older than Second_Baseman

## 1. Imort the dataset

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('../../data/SOCR_MLB.tsv', sep = '\t')
df

,Adam_Donachie,BAL,Catcher,74,180,22.99
0,Paul_Bako,BAL,Catcher,74,215.0,34.69
1,Ramon_Hernandez,BAL,Catcher,72,210.0,30.78
2,Kevin_Millar,BAL,First_Baseman,72,210.0,35.43
3,Chris_Gomez,BAL,First_Baseman,73,188.0,35.71
4,Brian_Roberts,BAL,Second_Baseman,69,176.0,29.39
...,...,...,...,...,...,...
1028,Brad_Thompson,STL,Relief_Pitcher,73,190.0,25.08
1029,Tyler_Johnson,STL,Relief_Pitcher,74,180.0,25.73
1030,Chris_Narveson,STL,Relief_Pitcher,75,205.0,25.19
1031,Randy_Keisler,STL,Relief_Pitcher,75,190.0,31.01


### Now, we'll rename the columns.

In [3]:
df.columns = ['Name', 'Team', 'Role', 'Height', 'Weight', 'Age']
df

,Name,Team,Role,Height,Weight,Age
0,Paul_Bako,BAL,Catcher,74,215.0,34.69
1,Ramon_Hernandez,BAL,Catcher,72,210.0,30.78
2,Kevin_Millar,BAL,First_Baseman,72,210.0,35.43
3,Chris_Gomez,BAL,First_Baseman,73,188.0,35.71
4,Brian_Roberts,BAL,Second_Baseman,69,176.0,29.39
...,...,...,...,...,...,...
1028,Brad_Thompson,STL,Relief_Pitcher,73,190.0,25.08
1029,Tyler_Johnson,STL,Relief_Pitcher,74,180.0,25.73
1030,Chris_Narveson,STL,Relief_Pitcher,75,205.0,25.19
1031,Randy_Keisler,STL,Relief_Pitcher,75,190.0,31.01


## 2. Calculate mean confidence interval

In [5]:
import scipy.stats

In [11]:
import numpy as np

In [12]:
def mean_confidence_interval(data, confidence=0.95):
    a = 1.0 * data
    n = len(a)
    m, se = np.mean(a), scipy.stats.sem(a)
    h = scipy.stats.t.ppf((1+confidence)/2, n-1) * se
    return m, h

In [17]:
df.groupby('Role').agg(
    Mean_Weight = ('Weight', 'mean'),
    Mean_Height = ('Height', 'mean'),
    Mean_Age = ('Age', 'mean')
)

,Mean_Weight,Mean_Height,Mean_Age
Role,,,
Catcher,204.653333,72.706667,29.651333
Designated_Hitter,220.888889,74.222222,30.389444
First_Baseman,213.109091,74.000000,29.474727
Outfielder,199.113402,73.010309,28.953454
Relief_Pitcher,203.517460,74.374603,28.539810
Second_Baseman,184.344828,71.362069,29.026897
Shortstop,182.923077,71.903846,28.402885
Starting_Pitcher,205.163636,74.719457,28.236109
Third_Baseman,200.955556,73.044444,28.691111


In [18]:
for p in [0.85, 0.9, 0.95]:
    m1, h1 = mean_confidence_interval(df.loc[df['Role'] == 'First_Baseman', ['Age']], p)
    m2, h2 = mean_confidence_interval(df.loc[df['Role'] == 'Second_Baseman', ['Age']], p)
    print(f"p: {p:.2f}, 1st_Baseman_Height: {m1 - h1[0]:.2f}..{m1 + h1[0]:.2f}, 2nd_Baseman_Height: {m2 - h2[0]:.2f}..{m2 + h2[0]:.2f}")

p: 0.85, 1st_Baseman_Height: 28.56..30.39, 2nd_Baseman_Height: 28.18..29.87
p: 0.90, 1st_Baseman_Height: 28.42..30.53, 2nd_Baseman_Height: 28.06..29.99
p: 0.95, 1st_Baseman_Height: 28.22..30.73, 2nd_Baseman_Height: 27.87..30.18


## 3. Student's t-test

In [19]:
from scipy.stats import ttest_ind

In [21]:
tval, pval = ttest_ind(df.loc[df['Role'] == 'First_Baseman', ['Age']], df.loc[df['Role'] == 'Second_Baseman', ['Age']], equal_var=False)
print(f"t-val: {tval[0]:.2f}, p-val: {pval[0]:.2f}")

t-val: 0.53, p-val: 0.60


> Since, p-val = 0.60 (greater than 0.05). It fails to reject the Null Hypothesis (H0)